spec/models/merchant_spec.rb

```diff
@@ -142,5 +142,30 @@
         expect(@merchant_1.items_ready_to_ship).not_to eq([@item_1, @item_4])
       end
     end
+
+    describe "#top_sales_day" do
+      it "returns day of highest sales revenue" do
+        merchant_2 = create(:merchant)
+        item_5 = create(:item, merchant: merchant_2)
+        invoice_7 = create(:invoice, created_at: "2024-01-23")
+        invoice_8 = create(:invoice, created_at: "2013-11-10")
+        create(:transaction, invoice: invoice_7)
+        create(:transaction, invoice: invoice_8)
+        invoice_item_1 = create(:invoice_item, item: item_5, invoice: invoice_7, quantity: 1, unit_price: 1000)
+        invoice_item_2 = create(:invoice_item, item: item_5, invoice: invoice_8, quantity: 2, unit_price: 1500)
+
+        merchant_3 = create(:merchant)
+        item_6 = create(:item, merchant: merchant_3)
+        invoice_9 = create(:invoice, created_at: "2022-03-08")
+        invoice_10 = create(:invoice, created_at: "2023-05-19")
+        create(:transaction, invoice: invoice_9)
+        create(:transaction, invoice: invoice_10)
```

Comment:
Duplicate the transactions of one that is supposed to be less, and see what I mean about duplicating data when joining the transactions table